From 26c3271cf64cc003f201af41198890a5750ed53e Mon Sep 17 00:00:00 2001
From: Shivaprasad Hongal <shongal@codeaurora.org>
Date: Thu, 12 Jul 2018 16:24:45 -0700
Subject: [PATCH 5/5] fs_mgr: Wrapped key support for FBE

Add "ice_wrapped_key_supported" support to be read from
fstab.

CRs-Fixed: 2288316

Change-Id: I01f634e8d464059791d731d2af690bdfd8008107
---
 fs_mgr/fs_mgr_fstab.cpp | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/fs_mgr/fs_mgr_fstab.cpp b/fs_mgr/fs_mgr_fstab.cpp
index 68cc530dd..0677e3d4b 100644
--- a/fs_mgr/fs_mgr_fstab.cpp
+++ b/fs_mgr/fs_mgr_fstab.cpp
@@ -116,12 +116,14 @@ static struct flag_list fs_mgr_flags[] = {
 #define EM_AES_256_HEH  4
 #define EM_SPECK_128_256_XTS 5
 #define EM_SPECK_128_256_CTS 6
+#define EM_ICE_WRAPPED_KEY_SUPPORTED 7
 
 static const struct flag_list file_contents_encryption_modes[] = {
     {"aes-256-xts", EM_AES_256_XTS},
     {"speck128/256-xts", EM_SPECK_128_256_XTS},
     {"software", EM_AES_256_XTS}, /* alias for backwards compatibility */
     {"ice", EM_ICE},              /* hardware-specific inline cryptographic engine */
+    {"ice_wrapped_key_supported", EM_ICE_WRAPPED_KEY_SUPPORTED}, /* ICE engine with wrapped key support */
     {0, 0},
 };
 
-- 
2.17.1

